### PR TITLE
Check for btrfs module availability

### PR DESCRIPTION
**Pull request by vojtechtrefny**

### CODE REVIEW

File: src/utils/module.c [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+/**
+ * bd_utils_module_error_quark: (skip)
+ */
+GQuark bd_utils_module_error_quark (void)
+{
+    return g_quark_from_static_string ("g-bd-utils-module-error-quark");
+}
+
+gboolean bd_utils_have_kernel_module (const gchar *module_name, GError **error) {
```
Comment (vpodzime) [inline src/utils/module.c]: Please add documentation for these new public functions and add them to the *docs/libblockdev-sections.txt* file.

File: src/plugins/btrfs.c [hunk 2]
```diff
@@ -128,6 +128,16 @@ gboolean bd_btrfs_check_deps () {
         ret = ret && status;
     }
 
+    status = bd_utils_have_kernel_module ("btrfs", &error);
+    if (!ret) {
+        if (error) {
+            g_warning("Failed to check for btrfs kernel module availability: %s" , error->message);
```
Comment (vpodzime) [inline src/plugins/btrfs.c]: Missing space.

File: src/plugins/btrfs.c [hunk 3]
```diff
@@ -128,6 +128,16 @@ gboolean bd_btrfs_check_deps () {
...
+            g_clear_error (&error);
+        } else
+            g_warning ("The 'btrfs' kernel module is not available.");
+    }
+    ret = ret && status;
+
```
Comment (vpodzime) [inline src/plugins/btrfs.c]: I'm afraid we also have to do this at the beginning of most of the *btrfs_* functions.

**Comment by vojtechtrefny** [comment]
Jenkins, test this please.

**Pull request merged**